### PR TITLE
🔒️(api) securing API with LTI token

### PR DESCRIPTION
**Pull request by wilbrdt**
## Purpose

API was unsecure and anyone could make request to it from a browser.

## Proposal

Securing it with a LTI token verification on each endpoint.

### CODE REVIEW

File: src/api/plugins/tdbp/warren_tdbp/api.py [hunk 1]
```diff
@@ -23,13 +26,14 @@
...
 @router.get("/window")
 async def get_sliding_window(
-    course_id: str = Query(
-        description="The course identifier on Moodle",
-    ),
-    until: Optional[date] = Query(
-        description="End date until when to compute the sliding window",
-    ),
-):
+    course_id: Annotated[str, Depends(get_lti_course_id)],
```
Comment (quitterie-lcs) [inline src/api/plugins/tdbp/warren_tdbp/api.py:29]: Can you keep the description ? It is used for API documentation

File: src/api/plugins/tdbp/warren_tdbp/api.py [hunk 2]
```diff
@@ -61,12 +65,11 @@ async def get_sliding_window(
 
 @router.get("/cohort")
 async def get_cohort(
-    course_id: str = Query(
-        description="The course identifier on Moodle",
-    ),
-    until: Optional[date] = Query(
-        description="End date until when to compute the sliding window",
-    ),
+    course_id: Annotated[str, Depends(get_lti_course_id)],
```
Comment (quitterie-lcs) [inline src/api/plugins/tdbp/warren_tdbp/api.py:68]: Ditto

File: src/api/plugins/tdbp/warren_tdbp/api.py [hunk 3]
```diff
@@ -95,27 +98,30 @@ async def get_cohort(
...
-    ),
-    totals: bool = Query(False, description="Flag to activate to compute total scores"),
-    average: bool = Query(
-        False, description="Flag to activate to compute average scores"
-    ),
-):
+async def get_scores(  # noqa: PLR0913
+    course_id: Annotated[str, Depends(get_lti_course_id)],
+    roles: Annotated[List[LTIRole], Depends(get_lti_roles)],
+    user_id: Annotated[str, Depends(get_lti_user_id)],
```
Comment (quitterie-lcs) [inline src/api/plugins/tdbp/warren_tdbp/api.py:104]: Ditto

File: src/api/plugins/tdbp/warren_tdbp/api.py [hunk 4]
```diff
@@ -152,25 +164,26 @@ async def get_scores(
...
-    student_id: Optional[str] = Query(
-        None, description="The student identifier on Moodle"
-    ),
-    average: bool = Query(
-        False, description="Flag to activate to compute average grades"
-    ),
-):
+    course_id: Annotated[str, Depends(get_lti_course_id)],
+    roles: Annotated[List[LTIRole], Depends(get_lti_roles)],
+    user_id: Annotated[str, Depends(get_lti_user_id)],
```
Comment (quitterie-lcs) [inline src/api/plugins/tdbp/warren_tdbp/api.py:169]: Ditto

File: src/api/plugins/tdbp/warren_tdbp/api.py (hunk 1 above)
Comment (wilbrdt) [reply src/api/plugins/tdbp/warren_tdbp/api.py:29]: `course_id` and `user_id` are no longer query parameters, but are internally taken from the JWT token forged from the LTI token.
I agree that the token should appear in the documentation though.
When reading the [documentation](https://fastapi.tiangolo.com/tutorial/dependencies/#integrated-with-openapi), and the [utils function](https://github.com/openfun/warren/blob/d64132c60a445d57cd86739df083061338577453/src/api/core/warren/utils.py#L21) on Warren side, the token should be appearing on the OpenAPI doc, but it does not and I'm not sure why...

Comment (wilbrdt) [reply src/api/plugins/tdbp/warren_tdbp/api.py:29]: Opened issue on Warren https://github.com/openfun/warren/issues/261

**Review by lunika — APPROVED** [review]

**Pull request merged**